dix/req.c

```diff
@@ -1678,11 +1711,22 @@ static bool dix_cas_rop_clink_cb(struct m0_clink *cl)
 		rop = crop->crp_parent->dr_rop;
 		rop->dg_completed_nr++;
 		M0_PRE(rop->dg_completed_nr <= rop->dg_cas_reqs_nr);
-		if (rop->dg_completed_nr == rop->dg_cas_reqs_nr) {
-			rop->dg_ast.sa_cb = dix_rop_completed;
-			rop->dg_ast.sa_datum = dreq;
+
+		if (dreq->dr_dtx != NULL) {
+			rop->dg_ast.sa_cb = dix_rop_one_completed;
+			rop->dg_ast.sa_datum = crop;
```

Comment:
please use rop-> dg_ast = {.sa_cb = dix_rop_one_completed, sa_datum = crop}.
rationale:
don't need to look at other fields of ast, as they will be nulified.